src/main/webapp/app/exercises/text/assess/text-submission-assessment.component.ts

```diff
@@ -54,6 +60,7 @@ export class TextSubmissionAssessmentComponent extends TextAssessmentBaseCompone
     noNewSubmissions: boolean;
     hasAssessmentDueDatePassed: boolean;
     correctionRound: number;
+    resultId?: number;
```

Comment:
Yo don't need to use `undefined`, since you're always checking for `resultId != 0` you can just initialize it to 0.

Reply:
```suggestion
    resultId: number;
```